### PR TITLE
Fix All Material UI Warnings

### LINKED ISSUE #6669: After upgrading Material UI Core to 4.12.2 Core components gives warning messages in dev console
After updating our backstage dependencies using `backstage-cli versions:bump` up to [release-2021-07-29](https://github.com/backstage/backstage/releases/tag/release-2021-07-29) I now get the following errors in my dev console:

-----

### theme.mixins.gutters() is deprecated
```
Material-UI: theme.mixins.gutters() is deprecated.
You can use the source of the mixin directly:

      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      [theme.breakpoints.up('sm')]: {
        paddingLeft: theme.spacing(3),
        paddingRight: theme.spacing(3),
      },
```
Associated with [packages/core-components/src/layout/Content/Content.tsx#L27](https://github.com/backstage/backstage/blob/4045eaab952fec349e443990330bd1ac5b2c9c18/packages/core-components/src/layout/Content/Content.tsx#L27)

-----

###  Warning: Failed prop type: The prop `justify` of `ForwardRef(Grid)` is deprecated
```
checkPropTypes.js:20 Warning: Failed prop type: The prop `justify` of `ForwardRef(Grid)` is deprecated. Use `justifyContent` instead, the prop was renamed.
    in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
    in WithStyles(ForwardRef(Grid)) (created by MultiSignInPage)
    in article (created by Content)
    in Content (created by MultiSignInPage)
    in div (created by Page)
    in ThemeProvider (created by Page)
    in Page (created by MultiSignInPage)
    in MultiSignInPage (created by SignInPage)
    in SignInPage (at App.tsx:45)
    in SignInPage (created by SignInPageWrapper)
    in SignInPageWrapper (created by AppRouter)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by AppRouter)
    in AppRouter (at App.tsx:126)
    in RoutingProvider (created by Provider)
    in CssBaseline (created by WithStyles(CssBaseline))
    in WithStyles(CssBaseline) (created by AppThemeProvider)
    in ThemeProvider (created by AppThemeProvider)
    in AppThemeProvider (created by Provider)
    in AppContextProvider (created by Provider)
    in ApiProvider (created by Provider)
    in Provider (at App.tsx:123)
```

This appears to be in multiple places https://github.com/backstage/backstage/search?q=justify

-----

### Warning: Failed prop type: Material-UI: `overlap="circle"` was deprecated. Use `overlap="circular"
```
checkPropTypes.js:20 Warning: Failed prop type: Material-UI: `overlap="circle"` was deprecated. Use `overlap="circular"` instead.
    in ForwardRef(Badge) (created by WithStyles(ForwardRef(Badge)))
    in WithStyles(ForwardRef(Badge)) (created by ForwardRef)
    in div (created by ForwardRef)
    in a (created by Link)
    in Link (created by NavLink)
    in NavLink (created by ForwardRef)
    in ForwardRef (created by SidebarSearchField)
    in div (created by SidebarSearchField)
    in SidebarSearchField (created by SidebarSearch)
    in SidebarSearch (at Root.tsx:61)
    in div (created by Sidebar)
    in div (created by Sidebar)
    in Sidebar (at Root.tsx:59)
    in div (created by SidebarPage)
    in SidebarPage (at Root.tsx:58)
    in Root (at App.tsx:127)
    in Route (created by AppRouter)
    in Routes (created by AppRouter)
    in SignInPageWrapper (created by AppRouter)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by AppRouter)
    in AppRouter (at App.tsx:126)
    in RoutingProvider (created by Provider)
    in CssBaseline (created by WithStyles(CssBaseline))
    in WithStyles(CssBaseline) (created by AppThemeProvider)
    in ThemeProvider (created by AppThemeProvider)
    in AppThemeProvider (created by Provider)
    in AppContextProvider (created by Provider)
    in ApiProvider (created by Provider)
    in Provider (at App.tsx:123)
    in App (at src/index.tsx:7)
```

Associated with [/packages/core-components/src/layout/Sidebar/Items.tsx#L170](https://github.com/backstage/backstage/blob/29e3c15ee5ef97a39ad71647e85d38a294c71a2a/packages/core-components/src/layout/Sidebar/Items.tsx#L170)

## Expected Behavior
Ideally we should have no warnings in the dev console (in a perfect world)

## Possible Solution

Replace with suggestions given in the warnings. 
I'm not too clear what to do in the case of `theme.mixins.gutters() is deprecated` as I haven't used this directly and documentation is sparse but it does appear this may be the solution https://github.com/mui-org/material-ui/pull/22109

```
-theme.mixins.gutters(),
+paddingLeft: theme.spacing(2),
+paddingRight: theme.spacing(2),
+[theme.breakpoints.up('sm')]: {
+  paddingLeft: theme.spacing(3),
+  paddingRight: theme.spacing(3),
+},

```

## Steps to Reproduce

1) Update backstage dependencies using `backstage-cli versions:bump` up to [release-2021-07-29](https://github.com/backstage/backstage/releases/tag/release-2021-07-29)
2) start app
3) login/navigate around
4) note warnings in console

## Context

Its not too bad but makes me a little twitchy and would be nice not to have warnings in the console

## Your Environment

- NodeJS Version (v14.17.1):
- Operating System and Version (MacOS):
- Browser Information: Chrome Version 92.0.4515.107 (Official Build) (x86_64)

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #6669

### CODE REVIEW

File: packages/core-components/src/layout/Content/Content.tsx [hunk 1]
```diff
@@ -24,7 +24,12 @@ const useStyles = makeStyles((theme: Theme) => ({
     minWidth: 0,
     paddingTop: theme.spacing(3),
     paddingBottom: theme.spacing(3),
-    ...theme.mixins.gutters({}),
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    [theme.breakpoints.up('sm')]: {
+    paddingLeft: theme.spacing(3),
```
Comment (timbonicus) [inline packages/core-components/src/layout/Content/Content.tsx:30]: Think there's some indentation missing here, did you [run Prettier](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#coding-guidelines)?

**Comment by benjdlambert** [comment]
Thanks for the contribution @Patil2099!

Few things we need here - the commit needs a DCO sign off, which you can do by doing `git commit --amend --signoff` and then `git push --force` again. 

Think you'll also need to run prettier on the files that you've changed to so it will format them properly.

**Review by timbonicus — COMMENTED** [review]
Ahh, this is a great @Patil2099 ! Super appreciate you taking the time to work through these warnings. Eager to get this in! 🎉

**Comment by Patil2099** [comment]
I think it is ready now @benjdlambert  @timbonicus. Let me know if you want any changes.

**Comment by benjdlambert** [comment]
@paulpach think there was a few more of the prettier bits but I think I fixed it up for you.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
@OrkoHunter / @iamEAP can I get a +1 here :)

**Review by iamEAP — APPROVED** [review]

**Pull request merged**